Convert to R 4.1 pipes

jashapiro opened the pull request:
This PR is mostly about converting to new-style R pipes throughout the repository, as discussed in #598. I did not include `purrr` updates here, as some of those may require a bit more change. I did, however, include a change to use the ggplot2 `vars()` function to select columns within a `facet_` function.

The other changes here are a number of whitespace changes (you might want to tell github to ignore those for your review) and a few minor `renv` updates: the updates there are to the `ashr` package and its dependencies to remove an error that I noticed when testing and has been solved in later versions of the package. This resulted, for some reason, in a change also to the `Repository` flag for most packages, so I made those all uniformly `CRAN`, and updated the primary CRAN mirror.

jashapiro commented:
I also want to note one place where I did make some changes to text, introducing the new pipe with a reference to the old pipe: 
https://github.com/AlexsLemonade/training-modules/blob/f18876046a3cc9af73cd152cbf760ef70859d461/intro-to-R-tidyverse/03-intro_to_tidyverse.Rmd#L181-L186

sjspielman reviewed (commented):
The `renv` changes look fine and `facet_` changes are all set. I want to catalog what I found with pipes to make sure we got it all. Here are the non-`live.Rmd` files that still have `%>%`, as identified with `grep -rl "%>%" . | grep -v live.Rmd`:


```
./RNA-seq/setup/exercises/01-clean_metadata_leukemia.R
./RNA-seq/setup/exercises/03-clean_metadata_medulloblastoma.R
./RNA-seq/exercise_01-exploratory_data_analysis.Rmd
./RNA-seq/exercise_02-bulk_rnaseq.Rmd
./scripts/spell-check.R
./scRNA-seq-advanced/diagrams/technical_merge_sce.png
./scRNA-seq-advanced/setup/PBMC_TotalSeqB/filter_cellranger_citeseq.R
./scRNA-seq-advanced/setup/rms/integrate_rms.R
./scRNA-seq-advanced/setup/rms/prepare_rms_library.R
./machine-learning/05-machine_learning_exercise.Rmd
./scRNA-seq/exercise_01-scrna_quant.Rmd
./scRNA-seq/setup/glioblastoma/make_count_matrix.R
./scRNA-seq/setup/glioblastoma/get_srp_table.R
./scRNA-seq/setup/glioblastoma/summarize_fastqc.R
./module-cheatsheets/intro-to-R-tidyverse-cheatsheet.md
./pathway-analysis/exercise_01-ora.Rmd
./pathway-analysis/exercise_03-gsva.Rmd
./pathway-analysis/exercise_02-gsea.Rmd
./intro-to-R-tidyverse/exercise_03b-intro_to_tidyverse.Rmd
./intro-to-R-tidyverse/exercise_03a-intro_to_tidyverse.Rmd
./intro-to-R-tidyverse/03-intro_to_tidyverse.Rmd
./intro-to-R-tidyverse/scripts/00-setup-intro-to-R.R
```

- `%>%` is not really a problem to keep in setup scripts (note, I didn't look at the code in those files directly), but I see that you did make some changes in other setup scripts so have a look at the ones listed above as needed for consistency,
- Exercise notebooks are not handled in this repo, so those pipes are ok for here.
- `./intro-to-R-tidyverse/03-intro_to_tidyverse.Rmd` doesn't have `%>%` in the _code_, just in the next to introduce `|>`.  
- It's probably ok to keep `%>%` in the cheatsheet, but we should also add `|>` to the cheatsheet as well. That could be a separate PR, up to you.
- We can remove the `magrittr` library load from the following:
    - `./RNA-seq/exercise_01-exploratory_data_analysis.Rmd`:
    - `./RNA-seq/exercise_02-bulk_rnaseq.Rmd`
- From `grep -n "%>%" ./machine-learning/05-machine_learning_exercise.Rmd`, can update these lines:
    ```
    152:summary_df %>%
    223:treatment_plot <- df_to_plot %>%
    ```

jashapiro commented:
First, let me note that I love that `%>%` was found in a `.png`.

I didn't update the machine learning code as that may need a more complete update; We haven't taught that module in a while, and I expect there are more changes there, so I didn't want to do it piecemeal.

I will update the cheatsheet for this module, and finish with the setup scripts: My search pattern had included only notebooks, which is why some of the setup scripts ended up modified..

jashapiro commented:
I updated all of the setup scripts to match, but left the ML workbooks alone. 

I also updated the cheatsheet, but I have not yet included the updated pdf, as I wanted to get feedback on wording first.

This should be ready for re-review.

sjspielman reviewed (approved):
LGTM! 🚀
